### PR TITLE
Specify Audit-ID to identify disruption requests in api audit logs

### PR DESCRIPTION
**Pull request by dgoodwin**
[TRT-888](https://issues.redhat.com//browse/TRT-888)

### CODE REVIEW

File: pkg/monitor/backenddisruption/disruption_backend_sampler.go [hunk 1]
```diff
@@ -325,6 +336,9 @@ func (b *BackendSampler) checkConnection(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	if b.auditID != "" {
+		req.Header.Set(audit.HeaderAuditID, b.auditID)
```
Comment (deads2k) [inline pkg/monitor/backenddisruption/disruption_backend_sampler.go:340]: audit ID needs to be unique for every request.  You probably want to use a UUID.

File: test/extended/util/disruption/controlplane/known_backends.go [hunk 2]
```diff
@@ -215,7 +217,8 @@ func createAPIServerBackendSampler(clusterConfig *rest.Config, disruptionBackend
 	if err != nil {
 		return nil, err
 	}
-	backendSampler.WithUserAgent("openshift-origin-external-backend-sampler")
+	backendSampler = backendSampler.WithUserAgent("openshift-origin-external-backend-sampler")
```
Comment (deads2k) [inline test/extended/util/disruption/controlplane/known_backends.go:220]: could you adjust this user-agent like this: https://github.com/openshift/origin/pull/27755/files#diff-ec9a13d07468752514f167c195090f13d8b7f9cff945a0f3fc43a01e1ce469b6R219 ?

File: pkg/monitor/backenddisruption/disruption_backend_sampler.go (hunk 1 above)
Comment (deads2k) [reply pkg/monitor/backenddisruption/disruption_backend_sampler.go:340]: Lower down, please store the audit IDs of failed requests and serialize the list of failed audit IDs in the disruption json file.

Comment (dgoodwin) [reply pkg/monitor/backenddisruption/disruption_backend_sampler.go:340]: I think including failed audit IDs in the disruption file needs a much more involved refactor and will have to wait, there's no place to put them as we're basing the whole system off intervals and using the same error message to build start/end times over multiple requests. 

For now I'm thinking just log the failed UUIDs and we'll have to correlate manually when debugging.

File: pkg/monitor/backenddisruption/disruption_backend_sampler.go [hunk 3]
```diff
@@ -480,8 +488,32 @@ func (b *disruptionSampler) produceSamples(ctx context.Context, interval time.Du
...
+				// TODO: don't merge this, randomly log some successful requests so we can ensure the audit ids are
+				// making it to the apiserver logs.
+				r := rand.Intn(100)
+				if r == 50 {
+					logrus.WithFields(logrus.Fields{
+						"backend": b.backendSampler.disruptionBackendName,
+						"type":    b.backendSampler.connectionType,
+						"auditID": uid,
+					}).Infof("disruption sample succeeded: %v", sampleErr)
+				}
```
Comment (dgoodwin) [inline pkg/monitor/backenddisruption/disruption_backend_sampler.go:515]: Ignore this, will be removed soon. :)

**Comment by dgoodwin** [comment]
Verified this seems to be working with temp code I'm about to remove:

time="2023-03-08T20:34:46Z" level=info msg="disruption sample succeeded: <nil>" auditID=c7b46f84-48e3-4445-a9af-24c0cf1a53e5 backend=kube-api type=new

❯ curl https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/27780/pull-ci-openshift-origin-master-e2e-gcp-ovn-upgrade/1633536916034949120/artifacts/e2e-gcp-ovn-upgrade/gather-extra/artifacts/audit_logs/kube-apiserver/ci-op-rq144hp6-03fd1-wlrxc-master-1-audit.log | grep c7b46f84-48e3-4445-a9af-24c0cf1a53e5

```
{"kind":"Event","apiVersion":"audit.k8s.io/v1","level":"Metadata","auditID":"c7b46f84-48e3-4445-a9af-24c0cf1a53e5","stage":"ResponseComplete","requestURI":"/api/v1/namespaces/default","verb":"get","user":{"username":"system:admin","groups":["system:masters","system:authenticated"]},"sourceIPs":["34.148.90.132"],"userAgent":"openshift-external-backend-sampler-new-kube-api","objectRef":{"resource":"namespaces","namespace":"default","name":"default","apiVersion":"v1"},"responseStatus":{"metadata":{},"code":200},"requestReceivedTimestamp":"2023-03-08T20:34:46.969015Z","stageTimestamp":"2023-03-08T20:34:46.972695Z","annotations":{"authorization.k8s.io/decision":"allow","authorization.k8s.io/reason":""}}
```

**Comment by dgoodwin** [comment]
Ready for review.

**Comment by dgoodwin** [comment]
/override ci/prow/e2e-aws-ovn-fips
/override ci/prow/e2e-aws-ovn-serial

I believe these failures are not related to this PR.

**Pull request merged**